thrust/thrust/system/omp/detail/scan.h

```diff
@@ -117,7 +119,13 @@ OutputIterator scan_impl(
   }
   else
   {
-    ::cuda::std::exclusive_scan(block_sums.begin(), block_sums.end(), block_sums.begin(), accum_t{}, wrapped_binary_op);
+    // For no init inclusive scan: exclusive_scan starting at second element with first element as init
+    if (num_threads > 1)
```

Comment:
I think this condition is always true, since we handle `num_threads <= 1` serially above. I think we can remove the `if` and just use the body directory. You can of course replace it by an assertion instead:
```suggestion
    _CCCL_ASSERT (num_threads > 1, "");
```